infra/modules/azure_federated_identity_with_github/id_iam.tf

```
@@ -0,0 +1,31 @@
+resource "azurerm_role_assignment" "ci_subscription" {
+  for_each = var.continuos_integration.roles == null ? toset([]) : var.continuos_integration.roles.subscription
+
+  scope                = var.subscription_id
+  role_definition_name = each.value
+  principal_id         = azurerm_user_assigned_identity.ci[0].principal_id
+}
+
+resource "azurerm_role_assignment" "cd_subscription" {
+  for_each = var.continuos_delivery.roles == null ? toset([]) : var.continuos_delivery.roles.subscription
```

To make sure to not trigger an error if will be passed a non-empty list while `cd` is disabled
```suggestion
  for_each = (!var.continuos_delivery.enable || var.continuos_delivery.roles == null) ? toset([]) : var.continuos_delivery.roles.subscription
```